From 0b16030adde530e70c94f71a06688ff395e32e1e Mon Sep 17 00:00:00 2001
Message-Id: <0b16030adde530e70c94f71a06688ff395e32e1e.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sun, 12 Oct 2014 10:30:02 +0530
Subject: [PATCH 16/75] REVERTME [VPG]: drm/i915/adf: Adjust get/put sequence
 to adf

This patch will align with WA patch submitted for S0ix to work.

commit 65af75893bdecbe20697ba95d27675e3eab87c38
Author: Sagar Kamble <sagar.a.kamble@intel.com>
Date:   Tue Sep 16 23:03:36 2014 +0530

    REVERTME [VPG]: drm/i915: Power gating display wells at the end of
    runtime & system suspend

For: GMINL-3547
Change-Id: I7d70e084367afef2a7a5bef9b77736535c624854
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_adf_wrapper.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_adf_wrapper.c b/drivers/gpu/drm/i915/i915_adf_wrapper.c
index 233c291..bb5c61f 100644
--- a/drivers/gpu/drm/i915/i915_adf_wrapper.c
+++ b/drivers/gpu/drm/i915/i915_adf_wrapper.c
@@ -53,6 +53,7 @@ static struct drm_i915_private *i915_adf_dev;
 void i915_adf_wrapper_init(struct drm_i915_private *dev_priv)
 {
 	i915_adf_dev = dev_priv;
+	intel_runtime_pm_get(dev_priv);
 }
 
 void i915_adf_wrapper_teardown(void)
@@ -82,6 +83,7 @@ void intel_adf_display_rpm_get(void)
 		return;
 
 	dev_priv = i915_adf_dev;
+	intel_runtime_pm_get(dev_priv);
 	intel_display_set_init_power(dev_priv, true);
 }
 EXPORT_SYMBOL(intel_adf_display_rpm_get);
@@ -94,7 +96,7 @@ void intel_adf_display_rpm_put(void)
 		return;
 
 	dev_priv = i915_adf_dev;
-	intel_display_set_init_power(dev_priv, false);
+	intel_runtime_pm_put(dev_priv);
 }
 EXPORT_SYMBOL(intel_adf_display_rpm_put);
 
-- 
1.7.9.5

